okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/IoExceptionResponseHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * (c) Copyright 2017 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting3.okhttp;
+
+import com.google.common.io.CharStreams;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import okhttp3.Response;
+
+enum IoExceptionResponseHandler implements ResponseHandler<IOException> {
+    INSTANCE;
+
+    @Override
+    public Optional<IOException> handle(Response response) {
+        if (response.isSuccessful() || response.code() == 101) {
```

Comment:
Could we give this magic number a name since it's not present in the usual `javax.ws.rs.core.Response.Status`?

Reply:
done